htlcswitch/interceptable_switch.go

```diff
@@ -35,75 +35,195 @@ type InterceptableSwitch struct {
 	// htlcSwitch is the underline switch
 	htlcSwitch *Switch
 
-	// fwdInterceptor is the callback that is called for each forward of
-	// an incoming htlc. It should return true if it is interested in handling
-	// it.
-	fwdInterceptor ForwardInterceptor
+	// intercepted is where we stream all intercepted packets coming from
+	// the switch.
+	intercepted    chan []*htlcPacket
+	resolutionChan chan *FwdResolution
+
+	interceptorRegistration chan ForwardInterceptor
+
+	requireInterceptor bool
```

Comment:
To prevent startup order issues, I think it is key to have this boolean here in the interceptable switch.